File: Itinere_3/Spring-Boot/owner_dog/src/main/java/com/example/owner_dog/controllers/DogController.java
package com.example.owner_dog.controllers;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;

import com.example.owner_dog.models.Dog;
import com.example.owner_dog.repositories.DogRepository;
import com.example.owner_dog.repositories.OwnerRepository;

@Controller
public class DogController {
    
    final DogRepository repo;
    final OwnerRepository ownerRepo;

    public DogController(DogRepository repo, OwnerRepository ownerRepo) {
        this.repo = repo;
        this.ownerRepo = ownerRepo;
    }

    @GetMapping("/dogs")
    String index(Model model) {
        model.addAttribute("dogs", repo.findAll());
        return "/dogs/index";
    }

    @GetMapping("/dogs/create")
    String create(Model model) {
        model.addAttribute("owners", ownerRepo.findAll());
        return "/dogs/create";
    }

    @PostMapping("/dogs/store")
    String store(Dog d) {
        repo.save(d);
        return "redirect:/dogs";
    }

    @GetMapping("/dogs/show")
    String show(Model model, Long id) {
        model.addAttribute("d", repo.findById(id).orElse(null));
        return "/dogs/show";
    }

    @GetMapping("/dogs/edit")
    String edit(Model model, Long id) {
        model.addAttribute("owners", ownerRepo.findAll());
        model.addAttribute("d", repo.findById(id).orElse(null));
        return "/dogs/edit";
    }

    @PostMapping("/dogs/update")
    String update(Dog d) {
        repo.save(d);
        return "redirect:/dogs";
    }

    @PostMapping("/dogs/destroy")
    String destroy(Long id) {
        repo.deleteById(id);
        return "redirect:/dogs";
    }
}

File: Itinere_3/Laravel/owner_dog/app/Http/Controllers/DogController.php
<?php

namespace App\Http\Controllers;

use App\Models\Dog;
use App\Models\Owner;
use Illuminate\Http\Request;

class DogController extends Controller
{
    /**
     * Display a listing of the resource.
     */
    public function index()
    {
        return view("dogs.index", ["dogs" => Dog::all()]);
    }

    /**
     * Show the form for creating a new resource.
     */
    public function create()
    {
        return view("dogs.create", ["owners" => Owner::all()]);
    }

    /**
     * Store a newly created resource in storage.
     */
    public function store(Request $request)
    {
        Dog::create($request->all());
        return redirect()->route("dogs.index");
    }

    /**
     * Display the specified resource.
     */
    public function show(Dog $dog)
    {
        return view("dogs.show", ["d" => $dog]);
    }

    /**
     * Show the form for editing the specified resource.
     */
    public function edit(Dog $dog)
    {
        return view("dogs.edit", ["d" => $dog, "owners" => Owner::all()]);
    }

    /**
     * Update the specified resource in storage.
     */
    public function update(Request $request, Dog $dog)
    {
        $dog->update($request->all());
        return redirect()->route("dogs.index");
    }

    /**
     * Remove the specified resource from storage.
     */
    public function destroy(Dog $dog)
    {
        $dog->delete();
        return redirect()->route("dogs.index");
    }
}
